### ISSUE TITLE
Segfault on Raspbian Jessie after upgrading to 0.6.1

### ISSUE THREAD

**Issue by booyaa** [rust-lang-nursery/rustup.rs#687]
system info:
```
booyaa@raspberrypi:~/stage $ uname -a
Linux raspberrypi 4.1.18-v7+ #846 SMP Thu Feb 25 14:22:53 GMT 2016 armv7l GNU/Linux
booyaa@raspberrypi:~/stage $ cat /etc/debian_version
8.0
```

Most commands seem to work though.

```
booyaa@raspberrypi:~/stage $ rustup -V
rustup 0.6.1 (71f7482 2016-08-24)
booyaa@raspberrypi:~/stage $ rustup
Segmentation fault
booyaa@raspberrypi:~/stage $ rustup show
Default host: armv7-unknown-linux-gnueabihf

installed toolchains
--------------------

stable-armv7-unknown-linux-gnueabihf
nightly-armv7-unknown-linux-gnueabihf (default)

active toolchain
----------------

nightly-armv7-unknown-linux-gnueabihf (default)
rustc 1.13.0-nightly (3c5a0fa45 2016-08-22)
```

**Comment by cutewalker** [comment]
yeah， I have run into really the same issue

system info:
 ```
$ uname -a
Linux raspberrypi 4.4.18-v7+ #905 SMP Fri Aug 19 19:11:00 BST 2016 armv7l GNU/Linux

$ cat /etc/debian_version
8.0

$ lsb_release -a
No LSB modules are available.
Distributor ID:	Raspbian
Description:   	Raspbian GNU/Linux 8.0 (jessie)
Release:       	8.0
Codename:      	jessie
```

also `rustup --version` and `rustup show` can work:
```
$ rustup --version
rustup 0.6.1 (71f7482 2016-08-24)

$ rustup show
Default host: armv7-unknown-linux-gnueabihf

stable-armv7-unknown-linux-gnueabihf (default)
rustc 1.11.0 (9b21dcd6a 2016-08-15)

$ rustup --help
Segmentation fault
```

**Comment by belstgut** [comment]
same problem

```
pi@babylon5 ~ % rustup --version
rustup 0.6.1 (71f7482 2016-08-24)
pi@babylon5 ~ % rustup show
Default host: armv7-unknown-linux-gnueabihf

installed toolchains
--------------------

stable-armv7-unknown-linux-gnueabihf (default)
nightly-armv7-unknown-linux-gnueabihf

active toolchain
----------------

stable-armv7-unknown-linux-gnueabihf (default)
rustc 1.11.0 (9b21dcd6a 2016-08-15)

pi@babylon5 ~ % rustup
[1]    31112 segmentation fault  rustup
139 pi@babylon5 ~ % uname -a                                                                                                                        :(
Linux babylon5 4.4.13-v7+ #894 SMP Mon Jun 13 13:13:27 BST 2016 armv7l GNU/Linux
pi@babylon5 ~ % cat /etc/os-release 
PRETTY_NAME="Raspbian GNU/Linux 8 (jessie)"
NAME="Raspbian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"

```

**Comment by brson** [comment]
Yuck. And this did not happen in 0.6.0? I _hope_ it's a regression in the nightly toolchain for armv7-unknown-linux-gnueabihf. I don't see any new bugs against that triple in the Rust repo though.

This may be a good reason to keep pinning rustup release builds to a specific toolchain instead of just building with whatever nightly.

I'm afraid I don't have any of this hardware, but if you are inclined you might try building rustup locally with the current nightly to reproduce, and then older nightlies to see if any of those don't cause the crash.

In the meantime you can get rustup 0.6 from https://static.rust-lang.org/rustup/dist/armv7-unknown-linux-gnueabihf/rustup-init.

**Comment by brson** [comment]
If you can verify that 0.6.0 works properly I will modify the release build scripts for travis to pin to a nightly around 8-22 and see if that fixes it.

**Comment by booyaa** [comment]
@brson Just tried rustup (from the link you suggested)

```
$ ./rustup-init -V
rustup-init 0.6.1 (71f7482 2016-08-24)
$ ./rustup-init
(installs options..)
$ rustup -V
rustup 0.6.1 (71f7482 2016-08-24)
 $ rustup
Segmentation fault
```
As a side note I noticed that rustup-init segfaults if you use `--help`

**Comment by brson** [comment]
Oh crap sorry. I gave you the wrong link. https://static.rust-lang.org/rustup/archive/0.6.0/armv7-unknown-linux-gnueabihf/rustup-init is 0.6.0

**Comment by brson** [comment]
Should be fixed, but I can't test this arch so waiting for confirmation.

**Comment by cutewalker** [comment]
yes, it is!  thanks very much!

**Comment by booyaa** [comment]
Still segfaulting (after downloading 0.6.0):

```
$ ./rustup-init -V
rustup-init 0.6.0 (2c38faa 2016-08-23)
```

**Comment by Boddlnagg** [comment]
@booyaa I think the fixed version is 0.6.2. Can you try that?

**Comment by booyaa** [comment]
@Boddlnagg Fixed in 0.6.2! Thank you!

**Comment by brson** [comment]
Great!
